include/zim/archive.h

```diff
@@ -106,7 +108,9 @@ namespace zim
       return other;
      }
 
+#ifdef LIBZIM_WITH_XAPIAN
      bool m_preloadXapianDb;
+#endif
```

Comment:
This makes the ABI dependent on the build options. As a result if an application compiled for one config of `libzim` happens to encounter in the wild a `libzim.so` of a different config a crash is very likely. I know that we already have this situation with libzim suggestions, but expanding it to `zim::OpenConfig` will make the potential problem much worse.